### PR TITLE
Fixing some docstrings that did not render as intended

### PR DESCRIPTION
**Pull request by dhermes**
Sphinx successfully rendered some docstrings that did things
different than intended. Found via visual inspection of a few
of the rendered files (more to come). Also

- using names of basic types (`dict` instead of `dictionary`,
  `bool` instead of `boolean`) since Sphinx recognizes them.
- fixing some broken Sphinx links
- fixing some broken URLs (404, docs moved)
- adding formatting to unformatted text
- renaming doc with underscores to use hyphen
- removing an accidentally public var that was documented (eklass -> _eklass)

### CODE REVIEW

File: gcloud/credentials.py [hunk 1]
```diff
@@ -46,38 +46,51 @@ def get_credentials():
...
-      which uses this method under the hood.
+
+        You should not need to use this function directly. Instead, use a
+        helper method which uses this method under the hood.
 
     Checks environment in order of precedence:
 
     * Google App Engine (production and testing)
-    * Environment variable GOOGLE_APPLICATION_CREDENTIALS pointing to
+    * Environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` pointing to
```
Comment (tseaver) [inline gcloud/credentials.py]: Maybe use the [`:envvar:` role](http://sphinx-doc.org/markup/inline.html#role-envvar)?

Comment (dhermes) [inline gcloud/credentials.py]: Neat! Didn't know it existed.

File: gcloud/connection.py [hunk 2]
```diff
@@ -305,12 +308,12 @@ def api_request(self, method, path, query_params=None,
...
                             gcloud-python.
 
-        :type expect_json: boolean
+        :type expect_json: bool
         :param expect_json: If True, this method will try to parse the
                             response as JSON and raise an exception if
                             that cannot be done.  Default is True.
 
-        :type _target_object: object or :class:`NoneType`
+        :type _target_object: :class:`object` or :class:`NoneType`
```
Comment (tseaver) [inline gcloud/connection.py]: Odd that `dict` and `bool` are understood, but not `object` (and even `None`/`NoneType`).

Comment (dhermes) [inline gcloud/connection.py]: It's the `or` that throws off Sphinx

File: gcloud/exceptions.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Custom exceptions for gcloud.storage package.
+"""Custom exceptions for :mod:`gcloud` package.
```
Comment (dhermes) [inline gcloud/exceptions.py]: FYI @tseaver I added this when updating to use `:envvar:`

**Comment by dhermes** [comment]
@tseaver PTAL I added `:envvar:` and rebased

**Comment by tseaver** [comment]
While I understand the desire for "pretty" history, this PR is a case where rebasing to squash commits interferes with review:  the small changes you just introduced would've been easier to re-review as separate commits.

LGTM

**Comment by dhermes** [comment]
Gothcer. In `oauth2client`, the maintainers usually ask for squashing after the fact. I should/can/will start doing that in PRs.

**Pull request merged**